engine/src/globals.cpp

```diff
@@ -928,8 +935,27 @@ bool X_open(int argc, MCStringRef argv[], MCStringRef envp[])
 				
 				tvar->setvalueref(*t_value);
 			}
-		}
+        }
 #endif // _SERVER
+
+    // SN-2015-07-17: [[ CommandArguments ]] Initialise the commandName and
+    //  commandArguments properties.
+    MCcommandname = NULL;
+    MCcommandarguments = NULL;
+    if (MCModeHasCommandLineArguments())
+    {
+        MCcommandname = MCValueRetain(argv[0]);
+
+        bool t_success;
+        t_success = MCArrayCreateMutable(MCcommandarguments);
+
+        // We build a 1-based numeric array.
+        for (int i = 1; t_success && i < argc; i++)
```

Comment:
The loop variable should be an `index_t`, to match the type of the argument to `MCArrayStoreValueAtIndex()`.